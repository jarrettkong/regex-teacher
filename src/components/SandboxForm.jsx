import React, { Component } from 'react';
import './SandboxForm.scss'

class SandboxForm extends Component {

  _handleForm = () => {
    const pattern = document.querySelector('#pattern-input').value;
    const flags = document.querySelector('#flag-input').value;
    this.props.handleForm(pattern, flags);
  }

  render() {
    return (
      <section>
        <form className="SandboxForm" onSubmit={this._handleSubmit}>
          <div className="Sandbox-input-container main-container">
            <label htmlFor="pattern-input">Pattern</label>
            <input type="text" id="pattern-input" spellCheck="false" placeholder="/pattern/" onChange={this._handleForm} minLength="1"/>
          </div>
          <div className="Sandbox-input-container flag-container">
            <label htmlFor="flag-input">Flags</label>
            <input type="text" id="flag-input" spellCheck="false" placeholder="/flags" onChange={this._handleForm} pattern="[gimsxeU]*"/>
          </div>
        </form>
        <hr/>
        <textarea name="Sandbox-textarea" id="Sandbox-textarea" className="Sandbox-textarea" rows="20" placeholder="Enter the text you want to test your regular expression against..." spellCheck="false" onChange={this.props.handleTextArea}></textarea>
      </section>
    )
  }
}

export default SandboxForm