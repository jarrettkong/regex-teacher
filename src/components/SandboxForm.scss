.SandboxForm {
  width: 100%;
  display: flex;
  
  & input {
    width: 100%;
    height: 40px;
    padding-left: 15px;
    font-size: 17px;
    border: 0;
    border-radius: 2px;
    margin-top: 5px;
    color: #DE6A73;
    font-family: 'Roboto Mono', monospace;

    &:focus {
      outline: transparent;
    }
  }
}

.main-container {
  flex: 8;
  margin-right: 30px;
}

.flag-container {
  flex: 2;
}

hr {
  margin: 40px 0;
}

.Sandbox-textarea {
  width: 100%;
  border: 0;
  border-radius: 2px;
  padding: 15px;
  font-size: 17px;
  &:focus {
    outline: transparent;
  }
}