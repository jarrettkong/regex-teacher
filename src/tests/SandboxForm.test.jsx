import React from 'react';
import SandboxForm from '../components/SandboxForm.jsx'
import { shallow } from 'enzyme';

const mockHandleForm = jest.fn();

describe('SandboxForm', () => {

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <SandboxForm handleForm={mockHandleForm} />
    )
  });

  it('should render the snapshot with all the data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('_handleForm', () => {

    it('should be called when pattern input changed', () => {
      wrapper.find('#pattern-input').simulate('change', { target: { value: 'pattern' } })
      expect(mockHandleForm).toHaveBeenCalled();
    })

    it('should be called when pattern input changed', () => {
      wrapper.find('#flag-input').simulate('change', { target: { value: 'flag' } })
      expect(mockHandleForm).toHaveBeenCalled();
    })

    it('should be called when pattern input changed', () => {
      wrapper.find('#pattern-input').simulate('change', { target: { value: 'pattern' } })
      expect(mockHandleForm).toHaveBeenCalled();
    })

  })

});