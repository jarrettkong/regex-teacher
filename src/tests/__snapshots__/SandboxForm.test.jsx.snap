// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SandboxForm should render the snapshot with all the data passed in 1`] = `
<section>
  <form
    className="SandboxForm"
  >
    <div
      className="Sandbox-input-container main-container"
    >
      <label
        htmlFor="pattern-input"
      >
        Pattern
      </label>
      <input
        id="pattern-input"
        minLength="1"
        onChange={[Function]}
        placeholder="/pattern/"
        spellCheck="false"
        type="text"
      />
    </div>
    <div
      className="Sandbox-input-container flag-container"
    >
      <label
        htmlFor="flag-input"
      >
        Flags
      </label>
      <input
        id="flag-input"
        onChange={[Function]}
        pattern="[gimsxeU]*"
        placeholder="/flags"
        spellCheck="false"
        type="text"
      />
    </div>
  </form>
  <hr />
  <textarea
    className="Sandbox-textarea"
    id="Sandbox-textarea"
    name="Sandbox-textarea"
    placeholder="Enter the text you want to test your regular expression against..."
    rows="20"
    spellCheck="false"
  />
</section>
`;
